packages/fxa-auth-db-mysql/lib/db/schema/patch-099-100.sql

```diff
@@ -0,0 +1,102 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('99');
+
+-- Once upon a time, we had a product name that was actually an ID.
+-- Now, we have both a human-readable name and an ID. Also, since
+-- the product name may not be unique, let's drop this index and
+-- replace it with a better one. We also haven't launched this
+-- feature yet, so we don't need to actually migrate any data
+
+ALTER TABLE `accountSubscriptions`
+  ADD COLUMN `productId` VARCHAR(191) DEFAULT NULL,
+  ALGORITHM = INPLACE, LOCK = NONE;
+
+DROP INDEX accountSubscriptionsUnique
+  ON `accountSubscriptions`
+  ALGORITHM = INPLACE LOCK = NONE;
+
+CREATE UNIQUE INDEX accountSubscriptionsUnique2
+  ON accountSubscriptions(uid, productId, subscriptionId);
+
+CREATE PROCEDURE `createAccountSubscription_2` (
+  IN inUid BINARY(16),
+  IN inSubscriptionId VARCHAR(191),
+  IN inProductId VARCHAR(191),
+  IN inProductName VARCHAR(191),
+  IN inCreatedAt BIGINT SIGNED
+)
+BEGIN
+
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET @accountCount = 0;
+
+  -- Signal error if no user found
+  SELECT COUNT(*) INTO @accountCount FROM accounts WHERE uid = inUid;
+  IF @accountCount = 0 THEN
+    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Can not create subscription for unknown user.';
+  END IF;
+
+  INSERT INTO accountSubscriptions(
+    uid,
+    subscriptionId,
+    productId,
+    productName,
```

Comment:
Hmm, I have mixed feelings about this.

On the one hand, it seems like in an ideal data model, `productName` would be normalised out to a separate `products` table. But I also get that these arrive as a pair from external sources outside our control, and we can't guarantee the consistency of `productName` over time. But then if `productName` were to suddenly change at some point for a given `productId`, wouldn't we want that change applied across the board? Or am I worrying about imaginary situations that will never occur in practice?

Reply:
Hmm, after sleeping on it, I'm tempted to trash this PR. Could just live with the ID being called "name" in the FxA database as it has been up til now. Renaming the column is really the only issue, since the purpose stays the same.

The main thing we care about in this table is taking the set of products granted to a user via subscriptions and translating those into capabilities for RPs.

The new `productName` is just basically a cached value. It's only ever displayed to the user on the subscriptions management page. But, the real source of truth for the name comes from subhub & Stripe APIs. And, we end up calling those on the management pages anyway.

Reply:
> The new `productName` is just basically a cached value. It's only ever displayed to the user on the subscriptions management page. But, the real source of truth for the name comes from subhub & Stripe APIs.

Fwiw, I can live with that explanation if you decide it's worth keeping. I'll r+ the PR so you have the option. :)

Reply:
> Renaming the column is really the only issue, since the purpose stays the same.

(or I could also live with a PR that renames to `productId` instead of adding a column, fwiw)